rfcs/text/0003-object-field.md

```diff
@@ -1,30 +1,33 @@
+Numerous SaaS and enterprise content/productivity services provide event data where the event reflects an action (typically `create`, `read`, `update`, `delete`) on what can broadly be considered items/documents/resources/records (originally dubbed "objects"). The are generally considered audit logs, or business analytics events. For some of these services, parts of (or in some cases, the entirety of) the corpus consists of _files_ (in the filesystem sense of the term) wherein the existing ECS `file` fields are adequate. However, most enterprise collaboration, storage, and communication systems (Workday, Salesforce, ServiceNow, Zoom, GitHub, G Suite, Zendesk, Jira, Confluence - Cloud - Server, etc.) present data units as "records", "documents", "tickets", "meetings" and more. ECS doesn't currently account for these entities, and relies on `file` as an option, albeit semantically incorrect and incomplete for the purpose of event tracking.
 
 ## Fields
 
 <!--
 Stage: 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
 -->
-This RFC calls for the introduction of a top-level `object` field with an initial fieldset of the following child fields:
+This RFC calls for the introduction of a top-level `enterprise_content` field with an initial fieldset of the following child fields:
 
 | field | type | description |
 | --- | --- | --- |
-| `object.name` | keyword | Name of the Object |
-| `object.id` | keyword | ID of the Object |
-| `object.type` | keyword | Type of Object Represented (e.g. `record`, `meeting`, `repository`, `organization`, etc.) |
-| `object.additional_details` | object | Custom Key/Value pairs representing data relevant to the object |
-| `object.owner.id` | keyword | ID of the Object Owner |
-| `object.owner.name` | keyword | Name of the Object Owner |
-| `object.owner.email` | keyword | Email of the Object Owner |
+| `enterprise_content.document.name` | keyword | Name of the Enterprise Content Object |
+| `enterprise_content.document.id` | keyword | ID of the Enterprise Content Object |
+| `enterprise_content.document.type` | keyword | Type of Enterprise Content Object represented (e.g. `record`, `meeting`, `repository`, `organization`, etc.) |
+| `enterprise_content.uri` | keyword | Absolute location of Enterprise Content Object  |
+| `enterprise_content.source.id` | keyword | ID for the Enterprise Content Object source |
+| `enterprise_content.source.name` | keyword | Name for the Enterprise Content Object source |
+| `enterprise_content.owner.id` | keyword | ID of the Enterprise Content Object Owner |
```

Comment:
ECS nests `user.*` in several places, and with [multiple user support] we can nest under a different name (nest user fields as `owner.*`).

Perhaps that could make sense here?

Reply:
Deferring to @ioanatia and @qhoxie on best approach.

Reply:
I'm not sure I completely understand the comment.  Is it that we can have multiple aliases for the `user.*` fields here to connote various roles like owner?  If so that sounds good to me.

Reply:
I didn't have [this reference link](https://github.com/elastic/ecs/blob/master/rfcs/text/0007-multiple-users.md) correct before in my original comment. 🤦 

ECS already makes use of `user.*` field reuse, with nestings such as `client.user`, `destination.user`, etc. Recently support was added to not only nest `user.*`, but nest the `user.*` fields under a different name.

The `enterprise_content` fieldset could keep the `enterprise_content.owner.*` naming and nest the  [`user.*` fields](https://www.elastic.co/guide/en/ecs/current/ecs-user.html) underneath: `enterprise_content.owner.id`, `enterprise_content.owner.name`, `enterprise_content.owner.email`, etc.